pkg/controller/dnsrecord/actuator.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dnsrecord
+
+import (
+	"context"
+	"fmt"
+
+	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
+	"github.com/gardener/gardener/extensions/pkg/controller/dnsrecord"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	extensionsv1alpha1helper "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1/helper"
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener-extension-provider-aws/pkg/aws"
+	awsclient "github.com/gardener/gardener-extension-provider-aws/pkg/aws/client"
+)
+
+type actuator struct {
+	client client.Client
+	logger logr.Logger
+}
+
+func NewActuator(logger logr.Logger) dnsrecord.Actuator {
+	return &actuator{
+		logger: logger.WithName("aws-dnsrecord-actuator"),
+	}
+}
+
+func (a *actuator) InjectClient(client client.Client) error {
+	a.client = client
+	return nil
+}
+
+// Reconcile reconciles the DNSRecord.
+func (a *actuator) Reconcile(ctx context.Context, dns *extensionsv1alpha1.DNSRecord, cluster *extensionscontroller.Cluster) error {
+	// Create AWS client
+	awsClient, err := aws.NewClientFromDNSSecretRef(ctx, a.client, dns.Spec.SecretRef, dns.Spec.Region)
+	if err != nil {
+		return fmt.Errorf("could not create AWS client: %+v", err)
+	}
+
+	// Determine DNS hosted zone ID
+	zone, err := a.getZone(ctx, dns, awsClient)
+	if err != nil {
+		return err
+	}
+
+	// Create or update DNS record
+	ttl := extensionsv1alpha1helper.GetDNSRecordTTL(dns.Spec.TTL)
+	a.logger.Info("Creating or updating DNS record", "zone", zone, "name", dns.Spec.Name, "type", dns.Spec.RecordType, "values", dns.Spec.Values)
```

Comment:
I don't remember the details: is the region an important detail in Route53? If so, I might make sense to add a respective field to this log message.

I don't see anything about regions in the dashboard and its not "regionized", so I suppose it's not important. But I don't know the details about the gov cloud use case.

Reply:
I also don't think the region is important. It's needed to create the client, but apart from that it doesn't seem to make a difference. I think the same records will be managed if the region changes. So I would avoid making it too prominent.

Reply:
Ok, good. Then let's keep it like it is.

Reply:
In fact, it seems the region is not used **at all**. I tried with regions `foo` and `bar` and everything still works ... Still, this is route53-specific behavior, so I believe we should keep the region in the API as it *might* be important on other infrastructures.

Reply:
Sounds good, thanks for double-checking.

Reply:
The region **is used** in some special cases:
1. "Route 53 in the Beijing and Ningxia Regions: specify cn-northwest-1." see https://docs.aws.amazon.com/general/latest/gr/r53.html
2. we make use of regions  for NS2 (AWS regional API access). It also needs another endpoint: see https://github.com/gardener/external-dns-management/blob/master/pkg/controller/provider/aws/handler.go#L99-L101
/cc @danielfoehrKn